### DISCUSSION TITLE
configure token_endpoint_auth_methods_supported

### DISCUSSION THREAD

**Discussion by karmaking** [ory/hydra#3219] [Q&A] [answered]
We need to disable the `none` method from `token_endpoint_auth_methods_supported` and `userinfo_signing_alg_values_supported` for a client. Is there some way to configure this?

Thanks and cheers!

**Comment by vinckr** [comment, marked as answer]
Hey @karmaking 
apologies for the late response, I am not sure if I can give you a good answer here.

The token_endpoint_auth_method parameter in the OAuth 2.0 client settings does allow you to specify the client authentication method for the token endpoint. The options include client_secret_basic, client_secret_post, private_key_jwt, and none. 

If you want to disable the none method, you might be able to do so by not including it in this parameter, but this is not explicitly stated in the documentation so I am not sure.

I recommend trying it out and reporting back 🙏

**Reply by karmaking** [reply]
Hi @vinckr , thanks for your reply!

I actually doubt it can be configured this way, because `TokenEndpointAuthMethodsSupported` seems hard-coded in https://github.com/ory/hydra/blob/82208c43a1a2e5d382db5ab35885b6b5042c9d54/oauth2/handler.go#L237 

Or am I missing the right place for this?
